src/plugins/fullUserInChatbox/index.tsx

```diff
@@ -0,0 +1,38 @@
+/*
+ * Vencord, a Discord client mod
+ * Copyright (c) 2024 Vendicated and contributors
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+import { Devs } from "@utils/constants";
+import definePlugin from "@utils/types";
+import { findComponentByCodeLazy } from "@webpack";
+
+const NormalMessageComponent = findComponentByCodeLazy(".USER_MENTION)");
+
+export default definePlugin({
+    name: "FullUserInChatbox",
+    description: "Puts the full user mention object in the chatbox",
+    authors: [Devs.sadan],
+
+    patches: [
+        {
+            find: ":\"text\":",
+            replacement: {
+                match: /(hidePersonalInformation.*?)return/,
+                replace: "$1return $self.patchChatboxMention(arguments[0]);"
+            }
+        }
+    ],
+
+    patchChatboxMention(props: any) {
```

Comment:
```suggestion
    UserMentionComponent(props: any) {
```

Also wrab in ErrorBoundary